### DISCUSSION TITLE
Issue mounting root filesystem in live cd

### DISCUSSION THREAD

**Discussion by piano1029** [dracutdevs/dracut#2606] [Q&A]
Hello,

I've made a live cd with this structure
- /boot
  - Kernel
  - Dracut initramfs
- /LiveOS
  - squashfs.img (squashfs image)
    - /LiveOS
      - rootfs.img (ext4 image)
        - Normal filesystem

Linux itself and dracut's first few parts boot fine, but when attempting to mount the root filesystem it goes bad:
```
[  OK  ] Found device /dev/mapper/live-rw.
[  OK  ] Finished dracut initqueue hook.
[  OK  ] Reached target Preparation for Remote File Systems.
[  OK  ] Reached target Remote File Systems.
         Starting dracut pre-mount hook...
[  OK  ] Finished dracut pre-mount hook.
         Mounting /sysroot...
[    5.598129] EXT4-fs (dm-0): mounted filesystem 60c51861-1d3d-43d9-bf2b-58d096309c03 r/w with ordered data mode. Quota mode: none.
[  OK  ] Mounted /sysroot.
[  OK  ] Reached target Initrd Root File System.
         Starting Mountpoints Configured in the Real Root...
[  OK  ] Finished Mountpoints Configured in the Real Root.
[  OK  ] Reached target Initrd File Systems.
[  OK  ] Reached target Initrd Default Target.
         Starting dracut mount hook...
[    5.638336] EXT4-fs (dm-0): unmounting filesystem 60c51861-1d3d-43d9-bf2b-58d096309c03.
[    5.668579] dracut-mount[525]: Warning: Can't mount root filesystem
         Starting Dracut Emergency Shell...
```

In the recovery shell I've mounted the the squashfs.img file to /squash and after that tried to mount the rootfs.img file using `mount /squash/LiveOS/rootfs.img /rootfs`, but this responds with `mount: /rootfs: /dev/loop1 already mounted or mount point busy.`, dmesg and journalctl don't show anything.

**Comment by LaszloGombos** [comment]
Please post your kernel/dracut command line. It should look similar to the test case dracut comes with - https://github.com/dracutdevs/dracut/blob/master/test/TEST-16-DMSQUASH/test.sh#L42

Perhaps the issue is that you do NOT need rootfs.img - squashfs.img should be able to contain a normal file system directly.

**Comment by piano1029** [comment]
My kernel command line was `root=live:/dev/sr0 console=ttyS0,9600`.

Running it with `rd.live.image rd.live.overlay.overlayfs=1 root=LABEL=ISOIMAGE` gives:
```
         Mounting /sysroot...
[    7.282487] overlayfs: failed to resolve '/run/overlayfs': -2
[FAILED] Failed to mount /sysroot.
See 'systemctl status sysroot.mount' for details.
```
The systemctl output is: `/sysroot: special device LiveOS_rootfs does not exist.`. Removing `rd.live.overlay.overlayfs=1` gives the same result as with the original options.

I had the squashfs image contain the root filesystem by itself before, but that gave the same results. I switched to the roofs.img approach as that is the traditional way according to dracut.cmdline(7)

**Reply by LaszloGombos** [reply]
> root=LABEL=

This is the label of the filesystem where the .iso file for the "live cd" is located.

**Reply by piano1029** [reply]
Okay, after using just `root=live:/dev/sr0 console=ttyS0,9600` and removing the rootfs.img file (so now everything inside of the squashfs file) it kinda works. sysroot.mount doesn't work but after creating the folders `/run/overlayfs` and `/run/ovlwork` manually it does. Should I have created those folders somewhere beforehand?

**Comment by piano1029** [comment]
@LaszloGombos By removing the rootfs.img layer and adding `rd.live.overlay.overlayfs=1` to the command line I get it to kinda boot, but it fails to mount the sysroot. When in the maintenance shell I can get the system to boot by running `systemctl start sysroot.mount` and killing the dracut-mount process.

In the dmesg log I see this error: `overlayfs: failed to resolve '/run/overlayfs': -2`
Checking the status of sysroot.mount doesn't show any errors.

How can I fix this sysroot mount issue?

**Comment by LaszloGombos** [comment]
@piano1029  You might be running into https://github.com/dracutdevs/dracut/issues/1820 . I would be valuable for the project (and also for you) to confirm. You do not need to remove systemd from your rootfs to try this - you can just disable including systemd for the generated initramfs.
